pkg/import/helpers_test.go

```diff
@@ -0,0 +1,209 @@
+		})
+	})
+
+	when("many registry credentials are provided by environment variables and Get is called", func() {
+		var credHelper *importpkg.CredHelper
+
+		it.Before(func() {
+			require.NoError(t, os.Setenv(envVarRegistryUrl+"_0", "fizz-registry.io"))
+			require.NoError(t, os.Setenv(envVarRegistryUser+"_0", "fizz-registry-user"))
+			require.NoError(t, os.Setenv(envVarRegistryPassword+"_0", "fizz-registry-password"))
+			require.NoError(t, os.Setenv(envVarRegistryUrl+"_1", "fuzz-registry.io"))
+			require.NoError(t, os.Setenv(envVarRegistryUser+"_1", "fuzz-registry-user"))
+			require.NoError(t, os.Setenv(envVarRegistryPassword+"_1", "fuzz-registry-password"))
+			require.NoError(t, os.Setenv(envVarRegistryUrl, "some-registry.io"))
+			require.NoError(t, os.Setenv(envVarRegistryUser, "some-registry-user"))
+			require.NoError(t, os.Setenv(envVarRegistryPassword, "some-registry-password"))
+		})
+
+		it.After(func() {
+			require.NoError(t, os.Setenv(envVarRegistryUrl+"_0", ""))
+			require.NoError(t, os.Setenv(envVarRegistryUser+"_0", ""))
+			require.NoError(t, os.Setenv(envVarRegistryPassword+"_0", ""))
+			require.NoError(t, os.Setenv(envVarRegistryUrl+"_1", ""))
+			require.NoError(t, os.Setenv(envVarRegistryUser+"_1", ""))
+			require.NoError(t, os.Setenv(envVarRegistryPassword+"_1", ""))
+			require.NoError(t, os.Setenv(envVarRegistryUrl, ""))
+			require.NoError(t, os.Setenv(envVarRegistryUser, ""))
+			require.NoError(t, os.Setenv(envVarRegistryPassword, ""))
+		})
+
+		it("returns username and password provided by environment variables", func() {
+			credHelper = importpkg.NewCredHelperFromEnvVars(envVarRegistryUrl, envVarRegistryUser, envVarRegistryPassword)
+			require.Equal(t, len(credHelper.Auths), 3)
+			registryUser, registryPassword, err := credHelper.Get(os.Getenv(envVarRegistryUrl + "_0"))
+			require.NoError(t, err)
+			require.Equal(t, "fizz-registry-user", registryUser)
+			require.Equal(t, "fizz-registry-password", registryPassword)
+			registryUser, registryPassword, err = credHelper.Get(os.Getenv(envVarRegistryUrl + "_1"))
+			require.NoError(t, err)
+			require.Equal(t, "fuzz-registry-user", registryUser)
+			require.Equal(t, "fuzz-registry-password", registryPassword)
+			registryUser, registryPassword, err = credHelper.Get(os.Getenv(envVarRegistryUrl))
+			require.NoError(t, err)
+			require.Equal(t, "some-registry-user", registryUser)
+			require.Equal(t, "some-registry-password", registryPassword)
+		})
+	})
+
+	when("more than 10 registry credentials are provided by environment variables that "+
+		"have a \"_N\" suffix (e.g., REGISTRY_URL_N where N is an integer >= 0 and <= 9) "+
+		"and no credential is provided by environment variables with an empty suffix (e.g., REGISTRY_URL) "+
+		"and Get is called", func() {
+		var credHelper *importpkg.CredHelper
+
+		it.Before(func() {
+			require.NoError(t, os.Setenv(envVarRegistryUrl+"_0", "zero-registry.io"))
```

Comment:
might be easier to use `os.UnsetEnv`

Reply:
Nice, that makes more sense.

Reply:
Addressed in commit https://github.com/vmware-tanzu/kpack-cli/pull/279/commits/36fc57bcc87b7a0b021ac4f14be32967c2772f77